### DISCUSSION TITLE
Missing RHEL 8 Python dependencies

### DISCUSSION THREAD

**Discussion by jw3** [ctc-oss/fapolicy-analyzer#414] [Developers]
We have a handful of dependencies that are not available on RHEL 8 for Python 3.9.

Need to identify the path to providing these.

@tparchambault is this correct and complete?

| Name                   | Has 3.6 RHEL package? | Has 3.9 Fedora package? |
|------------------------|:---------------------:|:-----------------------:|
| python3-gobject        |           Y           |            Y            |
| python3-cairo          |           Y           |            Y            |
| python3-rx             |           N           |            Y            |
| python3-events         |           N           |            Y            |
| python3-more-itertools |           Y           |            Y            |
| redux-py               |           N           |            N            |
| gtksourceview3    |     Y                  |         Y                |

- https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/package_manifest/appstream-repository

**Comment by tparchambault** [comment]
From our out-of-band email:

> All are not available in the RHEL8 Python3.9 packages.
> 
> I just verified your list against the default RHEL8 Python3.6 and believe it is accurate.
> 
> I don't have a FC34 system currently available, however I do know that Fedora 34 rpm installs work correctly out of the box, which implies your Fedora list is also correct, i.e. Fedora 34 has those packages available and dnf directly installs them as dependencies.

**Comment by jw3** [comment]
I added `gtksourceview3` to the list for completeness, it appears to be available.

**Comment by jw3** [comment]
Added bz reports for these

https://bugzilla.redhat.com/show_bug.cgi?id=2076803
https://bugzilla.redhat.com/show_bug.cgi?id=2076805

**Comment by tparchambault** [comment]
We also will need to add the epel dependency into the spec file (or to the README.md: yum -y install https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm) to address the following:
 <pre>
Installing dependencies:

python3-zipp                noarch 0.5.1-3.el8              epel          14 k

Installing weak dependencies:
 python3-configargparse      noarch 0.14.0-6.el8             epel          36 k
 python3-importlib-metadata  noarch 0.23-1.el8               epel          41 k
 python3-importlib-resources noarch 1.0.2-2.el8              epel          21 k
 python3-more-itertools      noarch 7.2.0-3.el8              epel          59 k
 </pre>

**Comment by jw3** [comment]
Fedora Testing repo available for events

https://bugzilla.redhat.com/show_bug.cgi?id=2082890

How to test:
https://fedoraproject.org/wiki/QA:Updates_Testing

**Comment by jw3** [comment]
They added rx to test repo

https://bugzilla.redhat.com/show_bug.cgi?id=2084778

**Comment by tparchambault** [comment]
Bottom Line:
Installed fapolicy-analyzer with epel-testing repo enabled. Rpm specified pkg dependencies were addressed via epel:

```
[toma@son-o-wimpy Archive]$ sudo dnf install ./fapolicy-analyzer-0.0.0-1.255.g1ed28e1.el8.x86_64.rpm 
Updating Subscription Management repositories.
Waiting for process with pid 3158 to finish.
Extra Packages for Enterprise Linux 8 - x86_64 - Source           218 kB/s | 2.5 MB     00:11    
Extra Packages for Enterprise Linux 8 - Testing - x86_64          1.5 MB/s | 877 kB     00:00    
Extra Packages for Enterprise Linux 8 - Testing - x86_64 - Source  67 kB/s |  29 kB     00:00    
Dependencies resolved.
==================================================================================================
 Package                         Arch       Version                        Repository        Size
==================================================================================================
Installing:
 fapolicy-analyzer               x86_64     0.0.0-1.255.g1ed28e1.el8       @commandline     811 k
Installing dependencies:
 python3-zipp                    noarch     0.5.1-3.el8                    epel              14 k
Installing weak dependencies:
 python3-configargparse          noarch     0.14.0-6.el8                   epel              36 k
 python3-events                  noarch     0.4-1.el8                      epel              21 k
 python3-importlib-metadata      noarch     0.23-1.el8                     epel              41 k
 python3-importlib-resources     noarch     1.0.2-2.el8                    epel              21 k
 python3-more-itertools          noarch     7.2.0-3.el8                    epel              59 k
 python3-rx                      noarch     3.2.0-3.el8                    epel             307 k

```

**Comment by tparchambault** [comment]
At runtime, a py 3.6 missing module error has been exposed on RHEL8.5. Currently investigating...

**Comment by jw3** [comment]
Updates picked back into the v0.5 branch.

https://github.com/ctc-oss/fapolicy-analyzer/tree/release-0.5.0

Release imminent.

**Reply by tparchambault** [reply]
Using epel-testing all dependencies resolved and installed over Rhel8.6. Depending on fapolicyd version, one may need to modify the `/root/.config/fapolicy-analyzer/fapolicy-analyzer.toml` config file to reference a different rules location. 

From:
```
rules_file_path = '/etc/fapolicyd/fapolicyd.rules'
```

To:
```
rules_file_path = '/etc/fapolicyd/rules.d'
```

**Reply by jw3** [reply]
> Using epel-testing all dependencies resolved and installed

The testing repo is no longer needed, these have been moved to the main EPEL repo.

**Reply by jw3** [reply]
I have now tested v0.5.2-rc1 on 8.6 with great success.

**Comment by tparchambault** [comment]
Also verified 0.5.2 rc1 rpm install over rhel8.6 w/epel repo enabled. All dependencies resolved via the rpm installation process; No runtime import exceptions.
